### PR TITLE
Full x2y2 rewrite

### PR DESCRIPTION
**Pull request by hildobby**
Brief comments on the purpose of your changes:

This PR helps:
- Make it a lot more efficient (runs in <1min for past week of data)
- Fixes some understated fees for when there are multiple recipients
- Overall should be a lot easier to iterate on when needed going forward
- Other small fixes that I have now forgotten as I type this cause I'm tired

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by 0xRobin** [comment]
@hildobby can you provide some example trades in the 'nft_trades_seed.csv' file?

**Comment by 0xRobin** [comment]
Thanks for addressing the issues @hildobby 
I'm going to push a few last tweaks and this should be good to merge today.

**Review by 0xRobin — APPROVED** [review]
Tweaks:
+ small refactoring changes to improve readability
  + extracted the sources, include project start date where possible
  + extracted the json logic
  + added some variables for static addresses
+ changed the price source to come from the inventory event instead of the profit event. The profit event price did not include the fees.
+ Some fixes in the seed file.

All tests are passing now :heavy_check_mark: 

@hildobby 1 small question about the private sale label logic, but other then that it's good to merge.

@jeff-dude FYI, this PR will change the `check_seed` generic test from warning to error

File: models/x2y2/ethereum/x2y2_ethereum_events.sql [hunk 1]
```diff
@@ -12,236 +12,188 @@
...
+, inv.price/POWER(10, currency_token.decimals) AS amount_original
+, pu.price*(inv.price/POWER(10, currency_token.decimals)) AS amount_usd
+, CASE WHEN inv.execution_delegate = '{{erc721_delegate}}' THEN 'erc721'
+    WHEN inv.execution_delegate = '{{erc1155_delegate}}' THEN 'erc1155'
     END AS token_standard
-, trade_type
+, 'Single Item Trade' AS trade_type
 , CAST(1 AS DECIMAL(38,0)) AS number_of_items
-, CASE WHEN et.`from`=seller THEN 'Offer Accepted'
+, CASE WHEN (inv.fees_0 IS NULL OR inv.fees_0_to != '{{fee_management_addr}}') AND (prof.evt_block_time < '2022-04-01' OR prof.evt_block_time >= '2022-05-01') THEN 'Private Sale'
```
[the remark this answers is not in the document]
Comment (jeff-dude) [reply models/x2y2/ethereum/x2y2_ethereum_events.sql:119]: sounds good on the switch to error, i almost pushed that on another PR. i think it's in a good spot to change and let PRs fail when seed test fail.
i'll wait here for response on private sales question, then we can look to merge

[the remark this answers is not in the document]
Comment (0xRobin) [reply models/x2y2/ethereum/x2y2_ethereum_events.sql:119]: Ok, then the current setup is clear to me.
> his check should imo be readded to both marketplace and royaty fees logic.

These checks are still there btw. 
https://github.com/duneanalytics/spellbook/blob/7eca58d0b0efe1e225a706b4f88c0a5cfbac3887/models/x2y2/ethereum/x2y2_ethereum_events.sql#L139-L141
https://github.com/duneanalytics/spellbook/blob/7eca58d0b0efe1e225a706b4f88c0a5cfbac3887/models/x2y2/ethereum/x2y2_ethereum_events.sql#L151-L153

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
merging since it passed tests prior to resolving merge conflicts with main. last commit had no changes to code.

**Pull request merged**